src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/DesignerTransaction.cs

```diff
@@ -81,51 +53,36 @@ public void Commit()
         }
 
         /// <summary>
-        ///     User code should implement this method to perform
-        ///     the actual work of committing a transaction.
+        /// User code should implement this method to perform the actual work of
+        /// committing a transaction.
         /// </summary>
         protected abstract void OnCancel();
 
         /// <summary>
-        ///     User code should implement this method to perform
-        ///     the actual work of committing a transaction.
+        /// User code should implement this method to perform the actual work of
+        /// committing a transaction.
         /// </summary>
         protected abstract void OnCommit();
 
         /// <summary>
-        ///     Overrides Object to commit this transaction
-        ///     in case the user forgot.
+        /// Overrides Object to commit this transaction in case the user forgot.
         /// </summary>
-        ~DesignerTransaction()
-        {
-            Dispose(false);
-        }
+        ~DesignerTransaction() => Dispose(false);
 
-        /// <internalonly/>
         /// <summary>
-        /// Private implementation of IDisaposable.
-        /// When a transaction is disposed it is
-        /// committed.
+        /// Private implementation of IDisaposable. When a transaction is disposed
+        /// it is committed.
         /// </summary>
         void IDisposable.Dispose()
         {
             Dispose(true);
 
-            // note - Dispose calls Cancel which sets this bit, so
-            //        this should never be hit.
-            //
```

Comment:
Why was this and the assert removed?

Reply:
Dispose is virtual, the second test here triggers the Debug.Fail

```cs
[Fact]
public void Dispose_FinalizeSuppressed_Success()
{
    var transaction = new NonDisposingDesignerTransaction();
    transaction.Cancel();
    ((IDisposable)transaction).Dispose();

    Assert.True(transaction.Canceled);
}

[Fact]
public void Dispose_FinalizeNotSuppressed_Success()
{
    var transaction = new NonDisposingDesignerTransaction();
    ((IDisposable)transaction).Dispose();

    Assert.False(transaction.Canceled);
}

private class NonDisposingDesignerTransaction : DesignerTransaction
{
    protected override void Dispose(bool disposing) { }

    protected override void OnCancel() { }
    protected override void OnCommit() { }
}
```